redis/connection.py

```diff
@@ -968,7 +976,33 @@ def _connect(self):
         sslsock = context.wrap_socket(sock, server_hostname=self.host)
         if self.ssl_validate_ocsp is True and CRYPTOGRAPHY_AVAILABLE is False:
             raise RedisError("cryptography is not installed.")
-        elif self.ssl_validate_ocsp is True and CRYPTOGRAPHY_AVAILABLE:
+
+        if self.ssl_ocsp_context is not None and self.ssl_validate_ocsp is not False:
+            raise RedisError(
+                "Either an OCSP context or validation may be specified " "but not both."
+            )
+
+        # validation for the stapled case
+        if self.ssl_ocsp_context:
```

Comment:
There should be an option for a user to not provide an OpenSSL context (nor an expected certificate as I said in other comments) and have redisPy create this temporary basic OpenSSL context for this OCSP staple verification.
This will make life easy for basic usage that simply asks for some staple verification, something like `request_ocsp_staple_verification=True` and that's it

Reply:
This will change in a future version - but given our use case, agreed.